networkit/cpp/graph/TopologicalSort.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * TopologicalSort.hpp
+ *
+ *  Created on: 22.11.2021
+ *      Author: Fabian Brandt-Tumescheit
+ */
+#include <networkit/graph/TopologicalSort.hpp>
+
+namespace NetworKit {
+
+TopologicalSort::TopologicalSort(const Graph &G) : G(&G) {
+    if (!G.isDirected()) {
+        throw std::runtime_error("Topological sort is defined for directed graphs only.");
+    }
+    n = G.numberOfNodes();
+    topSortMark.resize(n, NodeMark::NONE);
+    topology.resize(n, 0);
+    current = n - 1;
+}
+
+void TopologicalSort::run() {
+    if (hasRun)
+        reset();
+
+    std::stack<node> nodeStack;
+
+    G->forNodes([&](node u) {
+        if (topSortMark[u] != NodeMark::PERM) {
```

Comment:
Just a minor thing here: to avoid indenting the whole function, I would write something like
```
if (topSortMark[u] == NodeMark::PERM)
    return;
// Rest of the code
```